operators/validate_nid.go

```diff
@@ -83,21 +83,22 @@ func nidCl(nid string) bool {
 		}
 	}
 
-	val := 11 - (sum % 11)
+	val := sum % 11
 	switch val {
-	case 11:
+	case 0:
 		ndv = "0"
-	case 10:
+	case 1:
 		ndv = "k"
 	default:
-		ndv = strconv.Itoa(val)
+		ndv = strconv.Itoa(11 - val)
 	}
 	return ndv == dv
 }
 
+var nonDigit = regexp.MustCompile(`[^\d]`)
+
 func nidUs(nid string) bool {
-	re := regexp.MustCompile(`[^\d]`)
-	nid = re.ReplaceAllString(nid, "")
+	nid = nonDigit.ReplaceAllString(nid, "")
```

Comment:
Another optimization suggestion while we're here, since `strconv.Atoi` doesn't have a fast path for a single character, and also has logic for error cases though we've prevalidated, we can simplify around lines 114

`prev := int(nid[0] - '0')` ,`curr := int(nid[i] - '0')`